common/src/main/java/com/wynntils/features/map/WorldMarkersFeature.java

```diff
@@ -0,0 +1,539 @@
+ */
+package com.wynntils.features.map;
+
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.ByteBufferBuilder;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Services;
+import com.wynntils.core.consumers.features.Feature;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Category;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.persisted.config.ConfigCategory;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.event.RenderTileLevelLastEvent;
+import com.wynntils.mc.event.TickEvent;
+import com.wynntils.services.map.pois.WaypointPoi;
+import com.wynntils.services.mapdata.attributes.type.MapIcon;
+import com.wynntils.services.mapdata.attributes.type.ResolvedMapAttributes;
+import com.wynntils.services.mapdata.attributes.type.ResolvedMarkerOptions;
+import com.wynntils.services.mapdata.type.MapLocation;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.mc.type.Location;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import com.wynntils.utils.type.Pair;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.client.Camera;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.blockentity.BeaconRenderer;
+import net.minecraft.core.Position;
+import net.minecraft.world.phys.Vec2;
+import net.minecraft.world.phys.Vec3;
+import net.neoforged.bus.api.SubscribeEvent;
+import org.joml.Matrix4f;
+import org.joml.Quaternionf;
+import org.joml.Vector3d;
+import org.joml.Vector3f;
+import org.joml.Vector4f;
+
+@ConfigCategory(Category.MAP)
+public class WorldMarkersFeature extends Feature {
+    private static final float MINIMUM_RENDER_VISIBILITY = 0.1f;
+    private static final int RAINBOW_CHANGE_RATE = 10;
+
+    private static final MultiBufferSource.BufferSource BUFFER_SOURCE =
+            MultiBufferSource.immediate(new ByteBufferBuilder(256));
+
+    private static final WaypointPoi DUMMY_WAYPOINT = new WaypointPoi(() -> null, "");
+
+    @Persisted
+    public final Config<Float> textBackgroundOpacity = new Config<>(0.2f);
+
+    @Persisted
+    public final Config<Float> textRenderScale = new Config<>(1.0f);
+
+    @Persisted
+    public final Config<Float> iconRenderScale = new Config<>(1.0f);
+
+    @Persisted
+    public final Config<TextShadow> textShadow = new Config<>(TextShadow.NONE);
+
+    @Persisted
```

Comment:
These kind of conflicts with the attributes, yes? If you'd want to override those, should you just not have it as an overriding category with map attributes that ... hm... you can't really set a clipping limit there. Right. Ok, fine.